legend_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright ©2018 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plot
+
+import (
+	"image/color"
+	"os"
+	"testing"
+
+	"gonum.org/v1/plot/internal/cmpimg"
+	"gonum.org/v1/plot/vg"
+	"gonum.org/v1/plot/vg/draw"
+	"gonum.org/v1/plot/vg/vgimg"
+)
+
+type exampleThumbnailer struct {
+	color.Color
+}
+
+// Thumbnail fulfills the plot.Thumbnailer interface.
+func (et exampleThumbnailer) Thumbnail(c *draw.Canvas) {
+	pts := []vg.Point{
+		{c.Min.X, c.Min.Y},
+		{c.Min.X, c.Max.Y},
+		{c.Max.X, c.Max.Y},
+		{c.Max.X, c.Min.Y},
+	}
+	poly := c.ClipPolygonY(pts)
+	c.FillPolygon(et.Color, poly)
+
+	pts = append(pts, vg.Point{X: c.Min.X, Y: c.Min.Y})
+	outline := c.ClipLinesY(pts)
+	c.StrokeLines(draw.LineStyle{
+		Color: color.Black,
+		Width: vg.Points(1),
+	}, outline...)
+}
+
+// This example creates a some standalone legends with borders around them.
+func ExampleLegend_standalone() {
+
+	img := vgimg.New(vg.Points(120), vg.Points(100))
```

Comment:
I would call `img`, a value of type `*vgimg.Canvas`, `c` instead (to keep in line with other examples.)

but it's just nit-picking.